packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -43,28 +66,70 @@ class ColorScheme with Diagnosticable {
        assert(onSurface != null),
        assert(onBackground != null),
        assert(onError != null),
-       assert(brightness != null);
+       assert(brightness != null),
+       _primaryContainer = primaryContainer,
+       _onPrimaryContainer = onPrimaryContainer,
+       _secondaryContainer = secondaryContainer,
+       _onSecondaryContainer = onSecondaryContainer,
+       _tertiary = tertiary,
+       _onTertiary = onTertiary,
+       _tertiaryContainer = tertiaryContainer,
+       _onTertiaryContainer = onTertiaryContainer,
+       _errorContainer = errorContainer,
+       _onErrorContainer = onErrorContainer,
+       _outline = outline,
+       _surfaceVariant = surfaceVariant,
+       _onSurfaceVariant = onSurfaceVariant,
+       _inverseSurface = inverseSurface,
+       _inverseOnSurface = inverseOnSurface,
+       _inversePrimary = inversePrimary,
+       _shadow = shadow,
+       _primaryVariant = primaryVariant,
+       _secondaryVariant = secondaryVariant;
 
   /// Create a ColorScheme based on a purple primary color that matches the
   /// [baseline Material color scheme](https://material.io/design/color/the-color-system.html#color-theme-creation).
   const ColorScheme.light({
     this.primary = const Color(0xff6200ee),
-    this.primaryVariant = const Color(0xff3700b3),
-    this.secondary = const Color(0xff03dac6),
-    this.secondaryVariant = const Color(0xff018786),
-    this.surface = Colors.white,
-    this.background = Colors.white,
-    this.error = const Color(0xffb00020),
     this.onPrimary = Colors.white,
+    Color? primaryContainer,
+    Color? onPrimaryContainer,
+    this.secondary = const Color(0xff03dac6),
     this.onSecondary = Colors.black,
-    this.onSurface = Colors.black,
-    this.onBackground = Colors.black,
+    Color? secondaryContainer,
+    Color? onSecondaryContainer,
+    Color? tertiary,
+    Color? onTertiary,
+    Color? tertiaryContainer,
+    Color? onTertiaryContainer,
+    this.error = const Color(0xffb00020),
     this.onError = Colors.white,
+    Color? errorContainer,
+    Color? onErrorContainer,
+    Color? outline,
+    this.background = Colors.white,
+    this.onBackground = Colors.black,
+    this.surface = Colors.white,
+    this.onSurface = Colors.black,
+    Color? surfaceVariant,
+    Color? onSurfaceVariant,
+    Color? inverseSurface,
+    Color? inverseOnSurface,
+    Color? inversePrimary,
+    Color? shadow,
```

Comment:
Are we not adding m3 baselines from https://m3.material.io/styles/color/the-color-system/tokens at all? I think it'd be valuable to have a `lightm3` and `darkm3` constructors.

Reply:
I think they can be done in a follow-up (and also hooked into tokens once that flow is ready) but the value they add is possibly not a lot since its just a purple theme that users would change anyways (and if/when we add the ability to generate a color scheme from seed color, the need becomes even smaller)

Reply:
Fair point

Reply:
Yeah this was a point of discussion recently. It is not clear how useful these are to the average app developer and the naming becomes problematic.  That said, as Rami pointed out we could do this in another PR after we have an easier path for tokens to make it into the Flutter framework.